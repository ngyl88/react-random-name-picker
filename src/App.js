import React, { Component } from "react";
import math from "mathjs";
import Name from "./Name";
import RandomButton from "./RandomButton";
import InputForm from "./InputForm";

class App extends Component {
  constructor() {
    super();
    this.state = {
      names: ["gordon", "sahil", "david", "sally", "jane", "alice"],
      luckyWinnerIndex: null,
      inputName: "",
      isLoading: false
    };
  }

  render() {
    return (
      <div>
        <h1 id="app-title">hello random name picker</h1>
        {/* {this.state.isLoading ? console.log('loading...') : ''} */}
        <div><ul>
          {this.state.names.map((name, index) => (
            <Name
              key={index}
              name={name}
              isLucky={this.state.luckyWinnerIndex === index}
            />
          ))}
        </ul>
        <RandomButton handleClick={() => this.handleClick()} /></div>
        <InputForm
          inputName={this.state.inputName}
          handleNewName={event => this.handleNewNameChange(event)}
          handleSubmit={event => this.handleSubmit(event)}
        />
      </div>
    );
  }

  handleClick() {
    this.setState({
      isLoading: true
    });
    const randomIndex = math.randomInt(this.state.names.length);
    this.setState({
      luckyWinnerIndex: randomIndex
      // isLoading: false
    });
  }

  handleNewNameChange(event) {
    this.setState({
      inputName: event.target.value
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const newName = this.state.inputName;
    this.setState({
      names: [...this.state.names, newName],
      inputName: ""
    });
  }
}

export default App;
