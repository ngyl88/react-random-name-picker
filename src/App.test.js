import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import App from "./App";

// ES6 mocking
import { randomInt } from "mathjs";
jest.mock("mathjs", () => {
  return {
    randomInt: jest.fn()
  };
});

it("renders App without crashing", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});

it("renders App correctly after addNewName is called", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);

  const instance = renderer.getMountedInstance();
  instance.addNewName("Christmas");

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});

it("renders App correctly after handleClick is called", () => {
  // randomInt.mockReset();
  randomInt.mockImplementationOnce(num => 2);

  const renderer = new ShallowRenderer();
  renderer.render(<App />);

  const instance = renderer.getMountedInstance();
  instance.handleClick();

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});
