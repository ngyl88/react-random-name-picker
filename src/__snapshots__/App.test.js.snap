// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly after addNewName is called 1`] = `
<div>
  <h1
    id="app-title"
  >
    hello random name picker
  </h1>
  <div>
    <ul>
      <Name
        isLucky={false}
        name="gordon"
      />
      <Name
        isLucky={false}
        name="sahil"
      />
      <Name
        isLucky={false}
        name="david"
      />
      <Name
        isLucky={false}
        name="sally"
      />
      <Name
        isLucky={false}
        name="jane"
      />
      <Name
        isLucky={false}
        name="alice"
      />
      <Name
        isLucky={false}
        name="Christmas"
      />
    </ul>
    <RandomButton
      handleClick={[Function]}
    />
  </div>
  <InputForm
    addName={[Function]}
  />
</div>
`;

exports[`renders App correctly after handleClick is called 1`] = `
<div>
  <h1
    id="app-title"
  >
    hello random name picker
  </h1>
  <div>
    <ul>
      <Name
        isLucky={false}
        name="gordon"
      />
      <Name
        isLucky={false}
        name="sahil"
      />
      <Name
        isLucky={true}
        name="david"
      />
      <Name
        isLucky={false}
        name="sally"
      />
      <Name
        isLucky={false}
        name="jane"
      />
      <Name
        isLucky={false}
        name="alice"
      />
    </ul>
    <RandomButton
      handleClick={[Function]}
    />
  </div>
  <InputForm
    addName={[Function]}
  />
</div>
`;

exports[`renders App without crashing 1`] = `
<div>
  <h1
    id="app-title"
  >
    hello random name picker
  </h1>
  <div>
    <ul>
      <Name
        isLucky={false}
        name="gordon"
      />
      <Name
        isLucky={false}
        name="sahil"
      />
      <Name
        isLucky={false}
        name="david"
      />
      <Name
        isLucky={false}
        name="sally"
      />
      <Name
        isLucky={false}
        name="jane"
      />
      <Name
        isLucky={false}
        name="alice"
      />
    </ul>
    <RandomButton
      handleClick={[Function]}
    />
  </div>
  <InputForm
    addName={[Function]}
  />
</div>
`;
